### PR TITLE
feat: add diagnostic data and extra containers

### PR DESCRIPTION
**Pull request by Lash-L**
Adding diagnostic data that will allow users to get diagnostic data rather than having to use debug. Also adds stuff for flow led

### CODE REVIEW

File: roborock/roborock_typing.py [hunk 1]
```diff
@@ -306,6 +306,19 @@ class DockSummary:
     wash_towel_mode: Optional[WashTowelMode] = None
     smart_wash_params: Optional[SmartWashParams] = None
 
+    def as_dict(self) -> dict:
```
Comment (humbertogontijo) [inline roborock/roborock_typing.py]: Why not extend RoborockBase?

Comment (Lash-L) [reply roborock/roborock_typing.py]: changed

File: roborock/roborock_typing.py [hunk 2]
```diff
@@ -329,3 +330,13 @@ def update(self, device_prop: DeviceProp) -> None:
             self.last_clean_record = device_prop.last_clean_record
         if device_prop.dock_summary:
             self.dock_summary = device_prop.dock_summary
+
```
Comment (humbertogontijo) [inline roborock/roborock_typing.py]: Same here. We can use RoborockBase

Comment (Lash-L) [reply roborock/roborock_typing.py]: Changed.

**Review by humbertogontijo — APPROVED** [review]

**Pull request merged**